javaServices/coreJavaServices/src/joliex/io/FileService.java

```diff
@@ -434,15 +435,23 @@ private void writeXML(
 			return; // TODO: perhaps we should erase the content of the file before returning.
 		}
 		String rootName = value.children().keySet().iterator().next();
+		Value root = value.children().get( rootName ).get( 0 );
+		String rootNameSpace = "";
+		if ( root.hasChildren(NAMESPACE_ATTRIBUTE_NAME ) ) {
+			rootNameSpace = root.getFirstChild(  NAMESPACE_ATTRIBUTE_NAME ).strValue();
+		}
+		
 		try {
 			XSType type = null;
 			if ( schemaFilename != null ) {
 				try {
 					XSOMParser parser = new XSOMParser();
 					parser.parse( schemaFilename );
 					XSSchemaSet schemaSet = parser.getResult();
-					if ( schemaSet != null ) {
-						type = schemaSet.getElementDecl( "", rootName ).getType();
+					if ( schemaSet != null && schemaSet.getElementDecl( rootNameSpace, rootName ) != null ) {
+						type = schemaSet.getElementDecl( rootNameSpace, rootName ).getType();
+					} else if ( schemaSet.getElementDecl( rootNameSpace, rootName ) == null ) {
+						System.out.println("Root element " + rootName + " with namespace " + rootNameSpace + " not found in the schema " + schemaFilename );
```

Comment:
No println please. Is this an error? Should an exception be thrown?

Reply:
No, it is just a warning message